pydantic/types.py

```diff
@@ -1113,32 +1116,62 @@ def to(self, unit: str) -> float:
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ DATE TYPES ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 if TYPE_CHECKING:
+    from datetime import date
+
     PastDate = date
     FutureDate = date
 else:
 
-    class PastDate(date):
+    class PastDate(datetime.date):
         @classmethod
         def __get_validators__(cls) -> 'CallableGenerator':
             yield parse_date
             yield cls.validate
 
         @classmethod
-        def validate(cls, value: date) -> date:
-            if value >= date.today():
+        def validate(cls, value: datetime.date) -> datetime.date:
+            if value >= datetime.date.today():
                 raise errors.DateNotInThePastError()
 
             return value
 
-    class FutureDate(date):
+    class FutureDate(datetime.date):
         @classmethod
         def __get_validators__(cls) -> 'CallableGenerator':
             yield parse_date
             yield cls.validate
 
         @classmethod
-        def validate(cls, value: date) -> date:
-            if value <= date.today():
+        def validate(cls, value: datetime.date) -> datetime.date:
+            if value <= datetime.date.today():
                 raise errors.DateNotInTheFutureError()
 
             return value
+
+
+class ConstrainedDate(datetime.date, metaclass=ConstrainedNumberMeta):
+    gt: OptionalDate = None
+    ge: OptionalDate = None
+    lt: OptionalDate = None
+    le: OptionalDate = None
+
+    @classmethod
+    def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
+        update_not_none(field_schema, exclusiveMinimum=cls.gt, exclusiveMaximum=cls.lt, minimum=cls.ge, maximum=cls.le)
```

Comment:
can you add tests to check that schema is correct.

Specifically are dates encoded correctly in `schema_json()`?

Reply:
Sure - just trying to find an example to follow or test to modify...

Would this be the right place to do that?
https://github.com/pydantic/pydantic/blob/master/tests/test_schema.py#L459

Reply:
well probably best to do something like

```py
assert json.loads(model.schema_json()) == {
    ...
}
```
to avoid the test being brittle while testing the json data.

Reply:
only just saw this comment - thanks.  Will submit another change shortly to adjust